blocks/blogs/blogs.js

```diff
@@ -0,0 +1,118 @@
+import { callMockBlogAPI } from '../../scripts/mockapi.js';
+import { decorateIcons, fetchPlaceholders, readBlockConfig } from '../../scripts/aem.js';
+import { createPictureElement, listenToScroll } from '../../scripts/blocks-utils.js';
+
+async function createBlogCard(blogData) {
+  const ImageSrc = blogData.imageUrl;
+  const articleUrl = blogData.link;
+  const articleTitle = blogData.title;
+  const articleDesc = blogData.description;
+  const articleDate = blogData.postDate;
+
+  const arcticleDiv = document.createElement('div');
+  arcticleDiv.className = 'article';
+  const pictureWrapper = document.createElement('div');
+  pictureWrapper.className = 'picture-wrapper';
+  const articlePicture = createPictureElement(ImageSrc, 'article-thumbnail', false);
+
+  pictureWrapper.appendChild(articlePicture);
+  arcticleDiv.appendChild(pictureWrapper);
+  const articleContent = document.createElement('div');
+  articleContent.className = 'article-content';
+  const articleTitleDiv = document.createElement('h3');
+
+  const articleLink = document.createElement('a');
+  articleLink.href = articleUrl;
+  articleLink.textContent = articleTitle;
+  articleLink.target = '_blank';
+  articleTitleDiv.appendChild(articleLink);
+  articleContent.appendChild(articleTitleDiv);
+
+  const articleText = document.createElement('div');
+  articleText.className = 'article-text';
+
+  const articleTextP = document.createElement('span');
+  articleTextP.textContent = articleDesc;
+  articleText.appendChild(articleTextP);
+  articleContent.appendChild(articleText);
+  const articleInfo = document.createElement('div');
+  articleInfo.className = 'article-info';
+
+  const spanIcon = document.createElement('span');
+  spanIcon.className = 'icon icon-time';
+  articleInfo.appendChild(spanIcon);
+  decorateIcons(articleInfo, '');
+  const articleInfoTime = document.createElement('abbr');
+  articleInfoTime.textContent = articleDate;
+  articleInfo.appendChild(articleInfoTime);
+
+  const articleInfoPoweredBy = document.createElement('abbr');
+  articleInfoPoweredBy.textContent = (await fetchPlaceholders()).icicisecurities;
+  articleInfo.appendChild(articleInfoPoweredBy);
+  articleContent.appendChild(articleInfo);
+
+  arcticleDiv.appendChild(articleContent);
+  return arcticleDiv;
+}
+
+async function generateCardsView(block) {
+  const blogsContainer = block.querySelector('.blogs-cards-container');
+  const blogsDataArray = await callMockBlogAPI();
+  const entriesToProcess = blogsDataArray.length;
+  /**
+   * Loop through the blogsDataArray and create a blog card for each blog entry.
+   * Append the blog card to the blogsContainers column. Each column will have 2 blog cards.
+   */
+  // eslint-disable-next-line no-plusplus
+  for (let i = 0; i + 1 < entriesToProcess; i += 2) {
```

Comment:
How are we deciding about how many and which blog articles to show?

Reply:
Since original site is always having 10 blogs cards, I have mocked original data from website and created the same experience. We have to look this back when original API comes up. 
In Mobile - 1 column was shows, Tablet -3 and Desktop 5, have kept it same way